tests/approx_matching_ut.cpp

```diff
@@ -200,4 +200,52 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 			ACCEPTS("");
 		}
 	}
+
+	SIMPLE_UNIT_TEST(StressTest) {
+		ystring regexp = "^";
+		for (size_t letter = 0; letter < 10; ++letter) {
+			regexp += ystring(3, letter + 'a');
+		}
+		regexp += "$";
+
+		auto fsm = BuildFsm(regexp.data());
+
+		ystring changedRegexp = regexp.substr(1, regexp.size() - 2);
+
+		APPROXIMATE_SCANNER(fsm, 1) {
+			for (size_t pos = 0; pos < regexp.size() - 2; ++pos) {
+				changedRegexp[pos] = 'x';
```

Comment:
IMHO
It could be simpler to copy `text` into `changedText`, mutate that variable before a check and discard it afterwards.
Copy `text` again, mutate, check, discard, repeat.
That way it could be written as a for loop over operation id with a switch by that id:
```
for (int operation = 0; operation < 3; ++operation) {
        auto changedText = text;
        switch (operation) {
        case 0:
                changedText[pos] = 'x';
                break;
        ....
        }
        ACCEPTS(changedText);
```
Going deeper, the mutation can be made a function taking `text`, `pos` and `operation`.
It allows to reduce copy-paste at `APPROXIMATE_SCANNER(fsm, 2)`.